tests/lib/KeyStoreIndexeddb.spec.js

```diff
@@ -0,0 +1,140 @@
+        expect(KeyStore.instance._dbPromise).toBeNull();
+    });
+
+    it('can get a plain key', async () => {
+        const [key1, key2] = await Promise.all([
+            KeyStore.instance._getPlain(Dummy.keyInfo[0].userFriendlyAddress),
+            KeyStore.instance._getPlain(Dummy.keyInfo[1].userFriendlyAddress)
+        ]);
+        expect(key1).toEqual(Dummy.keyDatabaseEntries[0]);
+        expect(key2).toEqual(Dummy.keyDatabaseEntries[1]);
+    });
+
+    it('can get and decrypt a key', async () => {
+        const [key1, key2] = await Promise.all([
+            KeyStore.instance.get(Dummy.keyInfo[0].userFriendlyAddress, Dummy.encryptionPassword),
+            KeyStore.instance.get(Dummy.keyInfo[1].userFriendlyAddress, Dummy.encryptionPassword)
+        ]);
+        expect(key1.getPublicInfo()).toEqual(Dummy.keyInfo[0]);
+        expect(key2.getPublicInfo()).toEqual(Dummy.keyInfo[1]);
+        expect(Nimiq.BufferUtils.equals(key1.exportPlain(), Dummy.keyPairs[0])).toBe(true);
+        expect(Nimiq.BufferUtils.equals(key2.exportPlain(), Dummy.keyPairs[1])).toBe(true);
+    });
+
+    it('can get the type of a key', async () => {
+        const [type1, type2] = await Promise.all([
+            KeyStore.instance.getType(Dummy.keyInfo[0].userFriendlyAddress),
+            KeyStore.instance.getType(Dummy.keyInfo[1].userFriendlyAddress)
+        ]);
+        expect(type1).toBe(Dummy.keyInfo[0].type);
+        expect(type2).toBe(Dummy.keyInfo[1].type);
+    });
+
+    it('can list keys', async () => {
+        const keyInfo = await KeyStore.instance.list();
+        expect(keyInfo).toEqual(Dummy.keyInfo);
+    });
+
+    it('can remove keys', async () => {
+        let currentKeys = await KeyStore.instance.list();
+        expect(currentKeys).toEqual(Dummy.keyInfo);
+
+        await KeyStore.instance.remove(Dummy.keyInfo[0].userFriendlyAddress);
+        currentKeys = await KeyStore.instance.list();
+        expect(currentKeys.length).toBe(1);
+        expect(currentKeys[0].userFriendlyAddress).not.toBe(Dummy.keyInfo[0].userFriendlyAddress);
+
+        await KeyStore.instance.remove(Dummy.keyInfo[1].userFriendlyAddress);
+        currentKeys = await KeyStore.instance.list();
+        expect(currentKeys.length).toBe(0);
+
+        // check that we can't get a removed key by address
+        const removedKeys = await Promise.all([
+            KeyStore.instance._getPlain(Dummy.keyInfo[0].userFriendlyAddress),
+            KeyStore.instance._getPlain(Dummy.keyInfo[1].userFriendlyAddress)
+        ]);
+        expect(removedKeys[0]).toBeUndefined();
+        expect(removedKeys[1]).toBeUndefined();
+
+        // readd the keys to restore the state before the test
+        await Promise.all([
+            KeyStore.instance.putPlain(Dummy.keyDatabaseEntries[0]),
+            KeyStore.instance.putPlain(Dummy.keyDatabaseEntries[1])
+        ]);
+    });
+
+    it('can add and update keys', async () => {
+        // first clear database
```

Comment:
This test could be combined with the "delete" test, to avoid manually deleting and adding keys.

Reply:
thought about it but prefer testing them separately